File: python/Euclid_GCD.py
def gcd_recursive(a, b):
    """Recursively finds GCD of a and b."""
    if b == 0:
        return a
    else:
        return gcd_recursive(b, a % b)

def gcd_iterative(a, b):
    """Iteratively finds GCD of a and b."""
    while b:
        a, b = b, a % b
    return a

# --- Main Execution ---
if __name__ == "__main__":
    num1 = 48
    num2 = 18
    
    print(f"GCD of {num1} and {num2} (Recursive): {gcd_recursive(num1, num2)}")
    print(f"GCD of {num1} and {num2} (Iterative): {gcd_iterative(num1, num2)}")

File: java/EuclidGCD.java
public class EuclidGCD {

    // Recursive implementation
    public static int gcdRecursive(int a, int b) {
        if (b == 0) {
            return a;
        }
        return gcdRecursive(b, a % b);
    }

    // Iterative implementation (More stack-friendly)
    public static int gcdIterative(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    public static void main(String[] args) {
        int num1 = 48;
        int num2 = 18;

        System.out.println("GCD of " + num1 + " and " + num2 + " (Recursive): " + gcdRecursive(num1, num2));
        System.out.println("GCD of " + num1 + " and " + num2 + " (Iterative): " + gcdIterative(num1, num2));
    }
}